Kitodo-Validation/src/main/java/org/kitodo/validation/metadata/MetadataValidation.java

```diff
@@ -164,16 +165,18 @@ private static ValidationResult checkForStructuresWithoutMedia(Workpiece workpie
         boolean warning = false;
         Collection<String> messages = new HashSet<>();
 
-        Collection<String> structuresWithoutMedia = treeStream(workpiece.getRootElement(), IncludedStructuralElement::getChildren)
+        Collection<String> structuresWithoutMedia = treeStream(workpiece.getRootElement())
+                .filter(IncludedStructuralElement.class::isInstance).map(IncludedStructuralElement.class::cast)
```

Comment:
That’s a very general statement. What exactly do you think should be wrong about that?

At this point, the validation process checks whether each included structural element has also at least one media unit assigned. Now that included and linked structural elements can be contained in our root element tree, I have to look first to see if the subject under investigation is an included structural element at all. The linked structural elements cannot be associated with media units, I could not even call the appropriate getter, because linked structural elements do not have it. And that should not be reported as an error in the case either.

Do you think that would have to be implemented somehow differently—if so, how?